### PR TITLE
[RFC] Windows: restore defaults for 'shell' and 'shellcmdflags'

### PR DESCRIPTION
**Pull request by sethjackson**
On windows the default shell is `cmd.exe` and the shell flags `/c`.

Cherry picked from #810. equalsraf@2342660

### CODE REVIEW

File: src/nvim/options.lua [hunk 1]
```diff
@@ -2000,7 +2000,11 @@ return {
       vi_def=true,
       expand=true,
       varname='p_sh',
-      defaults={if_true={vi="sh"}}
+      defaults={
+        condition='WIN32',
+	if_true={vi="cmd.exe"},
+	if_false={vi="sh"}
```
Comment (splinterofchaos) [inline src/nvim/options.lua]: `condition` isn't aligned.

Comment (sethjackson) [inline src/nvim/options.lua]: `condition` is aligned I think (two spaces) but the `if_` aren't (used a tab). Correct?

Comment (justinmk) [inline src/nvim/options.lua]: We pretty much never use tabs for anything, though there might be some stragglers in some of our build scripts.

Comment (sethjackson) [inline src/nvim/options.lua]: Ok. I used all spaces in the updated diff.
Spaces are my preference too. :)

**Comment by justinmk** [comment]
Merged

**Pull request closed without merging**